### DISCUSSION TITLE
Unable to rebuild the bundle on local machine

### DISCUSSION THREAD

**Discussion by ashishprmodi** [AbsaOSS/spline-spark-agent#375] [Q&A]
Hi,

We are trying to build spline bundle after changing the lineage dispatcher. We are able to build the bundle. Once we deployed the bundle on our databricks cluster getting below mentioned error. 

"SecurityException: Invalid signature file digest for Manifest main attributes"

<img width="858" alt="image" src="https://user-images.githubusercontent.com/26406633/147197691-b2cb968e-d689-4bd0-8d51-f7589a87742d.png">

**Comment by wajda** [comment]
I bet it has something to do with the way you've made modifications to the bundle. I cannot tell for sure from that message alone, but it resembles the signed jar signature mismatch. If you repackaged the jar taken from Maven that would be the case. I can give the following suggestions:
1. Instead of rebuilding the bundle, consider packaging your custom extensions into a separate jar and submit it to Spark together with the original Spline agent bundle.
2. If you still decide to customize the original bundle then make it your artifact. Change the maven `groupId`, `atrifactId` or a `version` and build it from source. That would prevent any sort of collision with the original artifact (or signature).
